Sources/SourceControl/GitRepository.swift

```diff
@@ -203,14 +204,31 @@ public class GitRepository: Repository, WorkingCheckout {
     /// The path of the repository on disk.
     public let path: AbsolutePath
 
+    /// The (serial) queue to execute git cli on.
+    private let queue = DispatchQueue(label: "com.apple.swiftpm.gitqueue")
+
     public init(path: AbsolutePath) {
         self.path = path
     }
 
     // MARK: Repository Interface
 
-    public var tags: [String] { return tagsCache.getValue(self) }
-    private var tagsCache = LazyCache(getTags)
+    /// Returns the tags present in repository.
+    public var tags: [String] {
```

Comment:
Well... I actually think the ideal thing to expose in such situations is simply `AnySequence<String>` which is the most abstract type the API needs. That does imply exposing an order, though, even if its not part of the API contract.

The problem with using Set<> to "document" unorderedness is that it imposes an implementation restriction or performance penalty on the client if they don't want to keep it as a set.